test-suite/Math/NumberTheory/Moduli/PrimitiveRootTests.hs

```diff
@@ -90,6 +95,13 @@ isPrimitiveRootProperty4 (AnySign n) (Positive m)
     SomeMod n' -> not (isPrimitiveRoot n')
     InfMod{}   -> False
 
+isPrimitiveRootProperty5 :: Positive Natural -> Bool
+isPrimitiveRootProperty5 (Positive m)
+  = isNothing (cyclicGroupFromModulo m)
+  || case 0 `modulo` m of
+       SomeMod (_ :: Mod t) -> genericLength (filter isPrimitiveRoot [(minBound :: Mod t) .. maxBound]) == totient (totient m)
+       InfMod{}             -> False
+
```

Comment:
If you need to promote `m` to type level, there is a shorter approach: 

```haskell
case someNatVal m of
  SomeNat (_ :: Proxy t) -> ...
```

Reply:
Fixed.